memq/src/main/java/com/pinterest/memq/core/clustering/ExpirationPartitionBalanceStrategyWithErrorHandling.java

```diff
@@ -183,6 +190,41 @@ private void printRackStatus(Map<String, PriorityQueue<Broker>> rackBrokerCapaci
   protected abstract Set<Broker> handleBalancerError(Set<TopicConfig> topics, Set<Broker> brokers);
 
   protected void sendAlert() {
-    registry.counter(ALERT_METRIC).inc();
+    logger.info("Sending alert for balancer error.");
+    if (this.registry == null) {
+      try {
+        initializeMetricsRegistry();
+      } catch (Exception e) {
+        logger.severe("Failed to initialize metricsReporter. Cannot send alert." + e.getMessage());
+        return;
+      }
+    }
+    this.registry.counter(ALERT_METRIC).inc();
+  }
+
+  /**
+   * Initialize the metrics registry and the reporter.
+   * @throws Exception if the reporter cannot be initialized.
+   */
+  protected void initializeMetricsRegistry() throws Exception {
+    logger.info("Initializing metrics registry for balancer error.");
+    this.registry = new MetricRegistry();
+    String localHostname = MiscUtils.getHostname();
+    OpenTsdbConfiguration openTsdbConfiguration = this.memqConfig.getOpenTsdbConfig();
```

Comment:
can we add a null check to see if this.memqConfig is null? otherwise this might throw NPE if not set.

Reply:
is there any chance that `this.memqConfig.getOpenTsdbConfig()` can also return `null`? if so, that will also need a null check

Reply:
My plan is to add initializeMetricsRegistry into a try-catch (line 196) so it wont cause any crash if there's null value. 

Sure I can add null check here to provide better logging for debugging.